### PR TITLE
Change "Razor component(s)" terminology

### LINKED ISSUE #24912: Call out "Blazor component(s)" terminology
~Update instances to "Blazor component(s)" across the doc set.~

Actually, we're going to just make a quick remark that "Razor components" are informally known as "Blazor components."

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 81a32115-33f8-7d06-aaf0-b8ce918b5527
* Version Independent ID: 31ad18df-1904-bbdd-cfed-8d0d0267c10e
* Content: [ASP.NET Core Razor components](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/?view=aspnetcore-6.0)
* Content Source: [aspnetcore/blazor/components/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/components/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #24912

* This doesn't touch "Razor class library"/"RCL" references.

* Good news on the [Get-started/Build-your-first Lean Module](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro). It only uses the word "component(s)," so no changes are required to it.

* After this goes live, I'll ping a few external doc set authors (e.g., Azure docs) on instances outside of our doc set.

* *ei ei ei* ... There were even more of these than I thought. I thought that there were somewhere just north of 150 of these. Actually, there are ~360 of them across 60 files. I don't think we're going to really look 👀 at every one of these. File/folder names were **_all good_** ... NO changes to file/folder names are on this 🎉👍. For a couple of section headings that are changing, the build report will clear the cross-link updates that I made. Per the Olympic spirit ... **_Let's go for it!_** ⛷️🥇

* **_Wowzers!_** The PR built on the first shot without warnings/errors. The 🙏 **_servers gods_** 🙏 have blessed me! 😆

### CODE REVIEW

File: aspnetcore/blazor/call-web-api.md [hunk 1]
```diff
@@ -509,7 +509,7 @@ In `Program.cs`:
 builder.Services.AddHttpClient();
 ```
 
-The following Blazor Server Razor component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
+The following Blazor Server Blazor component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
```
Comment (danroth27) [inline aspnetcore/blazor/call-web-api.md:512]:
```suggestion
The following Blazor Server component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
```

File: aspnetcore/blazor/call-web-api.md [hunk 2]
```diff
@@ -519,7 +519,7 @@ The following Blazor Server Razor component makes a request to a web API for Git
 @using System.Text.Json.Serialization;
 @inject IHttpClientFactory ClientFactory
 
-<h1>Call web API from a Blazor Server Razor component</h1>
+<h1>Call web API from a Blazor Server Blazor component</h1>
```
Comment (danroth27) [inline aspnetcore/blazor/call-web-api.md:522]:
```suggestion
<h1>Call web API from a Blazor Server component</h1>
```

File: aspnetcore/blazor/call-web-api.md [hunk 3]
```diff
@@ -1124,7 +1124,7 @@ In `Startup.ConfigureServices` of `Startup.cs`:
 services.AddHttpClient();
 ```
 
-The following Blazor Server Razor component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
+The following Blazor Server Blazor component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
```
Comment (danroth27) [inline aspnetcore/blazor/call-web-api.md:1127]:
```suggestion
The following Blazor Server component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
```

File: aspnetcore/blazor/call-web-api.md [hunk 4]
```diff
@@ -1134,7 +1134,7 @@ The following Blazor Server Razor component makes a request to a web API for Git
 @using System.Text.Json.Serialization;
 @inject IHttpClientFactory ClientFactory
 
-<h1>Call web API from a Blazor Server Razor component</h1>
+<h1>Call web API from a Blazor Server Blazor component</h1>
```
Comment (danroth27) [inline aspnetcore/blazor/call-web-api.md:1137]:
```suggestion
<h1>Call web API from a Blazor Server component</h1>
```

**Comment by guardrex** [comment]
Indeed! @danroth27 ... I'll do a full search on that construct.

File: aspnetcore/blazor/call-web-api.md [hunk 5]
```diff
@@ -1741,7 +1741,7 @@ In `Startup.ConfigureServices` of `Startup.cs`:
 services.AddHttpClient();
 ```
 
-The following Blazor Server Razor component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
+The following Blazor Server Blazor component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
```
Comment (danroth27) [inline aspnetcore/blazor/call-web-api.md]:
```suggestion
The following Blazor Server component makes a request to a web API for GitHub branches similar to the *Basic Usage* example in the <xref:fundamentals/http-requests> article.
```

File: aspnetcore/blazor/call-web-api.md [hunk 6]
```diff
@@ -1751,7 +1751,7 @@ The following Blazor Server Razor component makes a request to a web API for Git
 @using System.Text.Json.Serialization;
 @inject IHttpClientFactory ClientFactory
 
-<h1>Call web API from a Blazor Server Razor component</h1>
+<h1>Call web API from a Blazor Server Blazor component</h1>
```
Comment (danroth27) [inline aspnetcore/blazor/call-web-api.md]:
```suggestion
<h1>Call web API from a Blazor Server component</h1>
```

**Review by danroth27 — APPROVED** [review]
LGTM

**Comment by guardrex** [comment]
@danroth27 A couple of more considerations ...

* The Razor SDK defines a target and a property ...

  `RazorComponentGenerate` ... defined as "Code generates *.cs* files for `RazorComponent` item elements. Use the `RazorComponentGenerateDependsOn` property to specify additional targets that can run before or after this target.

  Cross-references:

  * https://docs.microsoft.com/en-us/aspnet/core/razor-pages/sdk?view=aspnetcore-6.0
  * https://docs.microsoft.com/en-us/aspnet/core/razor-pages/sdk?view=aspnetcore-5.0
  * https://docs.microsoft.com/en-us/aspnet/core/razor-pages/sdk?view=aspnetcore-3.1
  * https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-6.0&tabs=visual-studio#changes-to-the-aspnet-core-razor-sdk
  
  I'll review them, but I'll be limited to just somehow adding (without disturbing props+targets) that they pertain to Blazor components.

* The `dotnet new` command uses `razorcomponent`. Can that receive an alias to include `blazorcomponent` for a future release? If so, I'll make a follow-up tracking entry to get it into the doc when it becomes available.

  Cross-ref: https://docs.microsoft.com/en-us/aspnet/core/tutorials/build-a-blazor-app?view=aspnetcore-6.0&pivots=server#build-a-todo-list-blazor-app

**Comment by danroth27** [comment]
Yeah, in the framework we try to avoid using product names that might change to align with the whims of marketing, especially when it might affect user code. You're correct that `RazorComponentGenerate` is for Razor/Blazor components. The Razor Component item template also shows up in VS with that name. The Razor SDK property and target names can't change, but we can probably rename/alias the item template (the Blazor project templates already have Blazor in their names). Let me know if you hit any other issues.

**Comment by guardrex** [comment]
That's ok ... it was just a giant find-'n-replace. It didn't take very long.

I think I should identify to you separately later tho those places externally that use "Blazor component" so that perhaps they can go inline with us here.

**Pull request closed without merging**